source/postprocess/mobility_statistics.cc

```diff
@@ -0,0 +1,157 @@
+/*
+  Copyright (C) 2011 - 2019 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#include <aspect/simulator.h>
+#include <aspect/postprocess/mobility_statistics.h>
+#include <aspect/material_model/simple.h>
+#include <aspect/global.h>
+#include <aspect/utilities.h>
+#include <aspect/geometry_model/interface.h>
+#include <aspect/geometry_model/spherical_shell.h>
+
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/fe/fe_values.h>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    template <int dim>
+    std::pair<std::string,std::string>
+    MobilityStatistics<dim>::execute (TableHandler &statistics)
+    {
+      // create a quadrature formula for the velocity for the volume of cells
+      const QGauss<dim> quadrature_formula (this->get_fe()
+                                            .base_element(this->introspection().base_elements.velocities).degree+1);
+      const unsigned int n_q_points = quadrature_formula.size();
+
+      FEValues<dim> fe_values (this->get_mapping(),
+                               this->get_fe(),
+                               quadrature_formula,
+                               update_values   |
+                               update_quadrature_points |
+                               update_JxW_values);
+
+      std::vector<Tensor<1,dim>> velocity_values(n_q_points);
+
+      // create a quadrature formula for the velocity for the surface of cells
+      const QGauss<dim-1> quadrature_formula_face (this->introspection().polynomial_degree.velocities+1);
```

Comment:
You probably want to introduce a similar variable like in the volume integration above:
```suggestion
      const QGauss<dim-1> quadrature_formula_face (this->introspection().polynomial_degree.velocities+1);
      const unsigned int n_q_points_face = quadrature_formula_face.size();
```